docs/pipeline.md

```diff
@@ -0,0 +1,161 @@
+      - my-log.json
+    output: json_parser
+  - type: json_parser
+    id: json_parser
+    output: stdout
+  - type: stdout
+    id: stdout
+```
+
+Additionally, we could accomplish the same task using custom `id`'s.
+
+```yaml
+pipeline:
+  - type: file_input
+    id: my_file
+    includes: 
+      - my-log.json
+    output: my_parser
+  - type: json_parser
+    id: my_parser
+    output: my_out
+  - type: stdout
+    id: my_out
+```
+
+We could even shuffle the order of operators, so long as we're explicitly declaring each output. This is a little counterintuitive, so it isn't recommended. However, it is shown here to highlight the fact that operators in a pipeline are ultimately connected via `output`'s and `id`'s.
+
+```yaml
+pipeline:
+  - type: stdout      # 3rd operator
+    id: my_out
+  - type: json_parser # 2nd operator
+    id: my_parser
+    output: my_out
+  - type: file_input  # 1st operator
+    id: my_file
+    includes: 
+      - my-log.json
+    output: my_parser
+```
+
+Finally, we could even remove some of the `id`'s and `output`'s, and depend on the default values. This is even less readable, so again would not be recommended. However, it is provided here to demonstrate that default values can be depended upon.
+
+```yaml
+pipeline:
+  - type: json_parser # 2nd operator
+  - type: stdout      # 3rd operator
+  - type: file_input  # 1st operator
+    includes: 
+      - my-log.json
+    output: json_parser
+```
+
+## Non-Linear Pipelines
+
+Now that we understand how `id` and `output` work together, we can configure `stanza` to run more complex pipelines. Technically, the structure of a `stanza` pipeline is limited only in that it must be a [directed, acyclic, graph](https://en.wikipedia.org/wiki/Directed_acyclic_graph).
+
+Let's consider a pipeline with two inputs and one output:
+```yaml
+pipeline:
+  - type: file_input
+    includes: 
+      - my-log.json
+    output: stdout # flow directly to stdout
+  - type: windows_eventlog_input
+    channel: security
+  - type: stdout
+```
+
+Here's another, where we read from two files that should be parsed differently:
+```yaml
+pipeline:
+  # Read and parse a JSON file
+  - type: file_input
+    includes: 
+      - my-log.json
+  - type: json_parser
+    output: stdout # flow directly to stdout
+  
+  # Read and parse a text file
+  - type: file_input
+    includes: 
+      - my-other-log.txt
+  - type: regex_parser
+    regex: ... # regex appropriate to file format
+    output: stdout
+
+  # Print
+  - type: stdout
+```
+
+Finally, in some cases, you might expect multiple log formats to come from a single input. This solution uses the [router](/docs/operators/router.md) operator. The `router` operator allows one to define multiple "routes", each of which has an `output`.
+
+
+```yaml
+pipeline:
+  # Read log file
+  - type: file_input
+    includes: 
+      - my-log.txt
+
+  # Route based on log type
+  - type: router
+    routes:
+      - expr: '$.looks_like == "format_one"'
```

Comment:
This is a little confusing because `$.looks_like` will never exist coming directly from a file input

Reply:
Might be better to do something like `- expr: '$record startsWith "prefix1"'`